File: tests/fixtures/python/df004_no_hitl.py
"""Fixture: should trigger DF004 (destructive @tool without human approval).

Each @tool function is named with a destructive verb (delete / send /
charge / deploy) and the file contains no HumanApprovalCallbackHandler,
no LangGraph interrupt_before, and no input(...) confirmation inside the
tool body. OWASP Agentic T10.

Suppresses DF001 / R001 with nemoguardrails + structlog so the golden
shows DF004 alone.
"""
import nemoguardrails  # noqa: F401  (suppresses DF001)
import structlog  # noqa: F401  (suppresses R001)

from langchain.tools import tool


@tool
def delete_user(user_id: str) -> str:  # DF004
    return f"deleted user {user_id}"


@tool
def send_email(recipient: str, body: str) -> str:  # DF004
    return f"sent email to {recipient}"


@tool("charge a customer card")
def charge_card(customer_id: str, amount_cents: int) -> str:  # DF004
    return f"charged {customer_id} ${amount_cents / 100:.2f}"


@tool
def deploy_release(version: str) -> str:  # DF004
    return f"deployed {version}"

File: agentshield/_retired_v2/fixtures/java/df004_no_hitl.java
// Fixture: should trigger DF004-java (destructive @Tool without human approval).
// Each method is named with a destructive verb (delete / send / charge /
// deploy) and the body has no confirm() / requireApproval() call.
package com.example.vuln;

import dev.langchain4j.agent.tool.P;
import dev.langchain4j.agent.tool.Tool;

public class NoHitlTools {

    @Tool("delete a user account")
    public String deleteUser(@P("user id") String userId) {  // DF004-java
        return "deleted user " + userId;
    }

    @Tool("send an email")
    public String sendEmail(@P("recipient") String to, @P("body") String body) {  // DF004-java
        return "sent email to " + to;
    }

    @Tool("charge a customer card")
    public String chargeCard(@P("customer id") String customerId, @P("amount in cents") int amount) {  // DF004-java
        return "charged " + customerId + " $" + (amount / 100.0);
    }

    @Tool("deploy a release")
    public String deployRelease(@P("version") String version) {  // DF004-java
        return "deployed " + version;
    }
}
